yt_dlp/extractor/ninenow.py

```diff
@@ -50,72 +62,84 @@ class NineNowIE(InfoExtractor):
+             lambda _, v: v['payload'][video_type]['slug'] == display_id,
+             'payload', any, {require('video data')}))
 
-        if not self.get_param('allow_unplayable_formats') and try_get(common_data, lambda x: x['episode']['video']['drm'], bool):
+        if traverse_obj(common_data, (video_type, 'video', 'drm', {bool})):
             self.report_drm(display_id)
-        brightcove_id = try_get(
-            common_data, lambda x: x['episode']['video']['brightcoveId'], str) or 'ref:{}'.format(common_data['episode']['video']['referenceId'])
-        video_id = str_or_none(try_get(common_data, lambda x: x['episode']['video']['id'])) or brightcove_id
-
-        title = try_get(common_data, lambda x: x['episode']['name'], str)
-        season_number = try_get(common_data, lambda x: x['season']['seasonNumber'], int)
-        episode_number = try_get(common_data, lambda x: x['episode']['episodeNumber'], int)
-        timestamp = unified_timestamp(try_get(common_data, lambda x: x['episode']['airDate'], str))
-        release_date = unified_strdate(try_get(common_data, lambda x: x['episode']['availability'], str))
-        thumbnails_data = try_get(common_data, lambda x: x['episode']['image']['sizes'], dict) or {}
-        thumbnails = [{
-            'id': thumbnail_id,
-            'url': thumbnail_url,
-            'width': int_or_none(thumbnail_id[1:]),
-        } for thumbnail_id, thumbnail_url in thumbnails_data.items()]
+        brightcove_id = traverse_obj(common_data, (
+            video_type, 'video', (
+                ('brightcoveId', {str}),
+                ('referenceId', {str}, {lambda x: f'ref:{x}' if x else None}),
+            ), any, {require('brightcove ID')}))
 
         return {
             '_type': 'url_transparent',
-            'url': smuggle_url(
-                self.BRIGHTCOVE_URL_TEMPLATE % brightcove_id,
-                {'geo_countries': self._GEO_COUNTRIES}),
-            'id': video_id,
-            'title': title,
-            'description': try_get(common_data, lambda x: x['episode']['description'], str),
-            'duration': float_or_none(try_get(common_data, lambda x: x['episode']['video']['duration'], float), 1000),
-            'thumbnails': thumbnails,
-            'ie_key': 'BrightcoveNew',
-            'season_number': season_number,
-            'episode_number': episode_number,
-            'timestamp': timestamp,
-            'release_date': release_date,
+            'ie_key': BrightcoveNewIE.ie_key(),
+            'url': self.BRIGHTCOVE_URL_TEMPLATE.format(brightcove_id),
+            **traverse_obj(common_data, {
+                'id': (video_type, 'video', 'id', {int}, ({str_or_none}, {value(brightcove_id)}), any),
+                'title': (video_type, 'name', {str}),
+                'description': (video_type, 'description', {str}),
+                'duration': (video_type, 'video', 'duration', {float_or_none(scale=1000)}),
+                'series': ('tvSeries', 'name', {str}),
```

Comment:
We can also add tags:
```suggestion
                'tags': (video_type, 'tags', ..., 'name', {str}),
                'series': ('tvSeries', 'name', {str}),
```